### PR TITLE
fix: change log file to json

### PR DESCRIPTION
**Pull request by mathild3r**
Changing the exist .log file into a json file
the ticket are added to an array of ticket. 
when all ticket are created => add the array to an interface with project ID associated, add this to a logFile interface and write to the file only at the end.
Difference from before: if the programme is killed the tickets send for the current project are not added to the file (the file is written too late) but they will if the programme failed. 

Screenshot 2021-11-18 at 14 28 44

### CODE REVIEW

File: jira_utils.go [hunk 1]
```diff
@@ -1,61 +1,22 @@
...
-	d := fmt.Sprint(now.Day()) + "_"
-	h := fmt.Sprint(now.Hour()) + "_"
-	min := fmt.Sprint(now.Minute()) + "_"
-	s := fmt.Sprint(now.Second())
-
-	return y + m + d + h + min + s
+type Tickets struct {
+	Summary         string `json:Summary`
+	Description     string `json:Description`
+	JiraIssueDetail string `json:jiraDetails`
```
Comment (aarlaud) [inline jira_utils.go:15]: this type feels a bit wrong/simplistic.... Why don't we have a type to know we expect key:Array[{something something}] ?

Comment (mathild3r) [reply jira_utils.go:15]: That 's just the structure for a ticket not for the full list (after the retry in the openJiraTickets). Maybe should be ticket not tickets. I guess you mean that you expect an other struct with, but I had an issue with that as it seem that the using a structure fix the key name and the key I use is the projects ID.

File: fixtures/results/logFileFixture.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (aarlaud) [inline fixtures/results/logFileFixture.json:1]: Try to prettify the json, makes it easier to read.

File: jira_utils.go [hunk 3]
```diff
@@ -1,61 +1,59 @@
...
-	}
-	defer f.Close()
+type JiradetailForTicket struct {
+	JiraIssue JiraIssueForTicket `json:jiraIssue`
+}
+
+type Tickets struct {
+	Summary         string `json:Summary`
+	Description     string `json:Description`
+	JiraIssueDetail map[string]interface{}
```
Comment (aarlaud) [inline jira_utils.go:25]: I think an interface is still too generic here. Need something more defined, like another Struct defining the fields that go in JiraIssueDetail.

Comment (mathild3r) [reply jira_utils.go:25]: the only way I can do that is if I decide not not put the issue ID in the JiraIssueDetail. Maybe I can do something like type Tickets struct {
	Summary         string `json:Summary`
	Description     string `json:Description`
        IssueId             string 'json:Id'
	JiraIssue        JiraIssue 'json:jiraIssue'
} with jira issue containing JiraIssue  ID and Key ?

Comment (aarlaud) [reply jira_utils.go:25]: can't we have IssueId being optional?

**Review by aarlaud — APPROVED** [review]

**Pull request merged**